### PR TITLE
[PLAT-973] Crossref is disabled with 'qatest' tag

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Currently we are minting/syncing new DOI and DOI metadata for QA activities. This fix enables QA to disable syncing with the `qatest` tag.

## Changes

- Adds check for tag before instantiating the client
- Adds new test

## QA Notes

add the qatest tag to the node of your preprint, go to https://test.crossref.org/servlet/submissionAdmin?sf=search and see that it does not update itself to sync with the preprint. Remove the tag and see that updating resumes.

## Documentation

Not a user facing change.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/projects/PLAT/issues/PLAT-973

### CODE REVIEW

**Comment by pattisdr** [comment]
Looks good @Johnetordoff

File: osf/models/preprint_service.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def set_identifier_values(self, doi, save=False):
             self.save()
 
     def get_doi_client(self):
-        if settings.CROSSREF_URL:
+        if settings.CROSSREF_URL and 'qatest' not in self.node.all_tags.values_list('name', flat=True):
```
Comment (sloria) [inline osf/models/preprint_service.py]: This should be able to be made into a query: `all_tags.filter(name='qatest').exists()`.

**Comment by Johnetordoff** [comment]
@sloria For `should_request_identifiers`,  I'm not sure if we want to instantiate the client at all if aren't going to do anything with it. Datacite will make an external call if it's instantiated to check it's credentials.  Also If a client is made and request_identifiers returns False that's another state we have to accommodate write tests around etc. I'm worried about creepin'.

**Comment by sloria** [comment]
> For should_request_identifiers, I'm not sure if we want to instantiate the client at all if aren't going to do anything with it. 

In my code snippet above, a client wouldn't be instantiated. To be more clear:

```python
def request_identifiers(target_object):
   #...
    if not target_object.should_request_identifiers():
        return False
    client = target_object.get_doi_client()
    if not client:
        return
```

`should_request_identifiers()` would check for the qatest tag.

File: tests/identifiers/test_datacite.py [hunk 2]
```diff
@@ -203,3 +206,9 @@ def test_datacite_get_by_identifier_not_found(self):
                 expect_errors=True,
             )
         assert res.status_code == 404
+
+    def test_qatest_doesnt_make_dois(self):
+        self.node.add_tag('qatest', auth=Auth(self.user))
+        assert not request_identifiers(self.node)
+
```
Comment (sloria) [inline tests/identifiers/test_datacite.py]: Nit: Remove blank lines at the ends of files.

**Pull request merged**